crates/red_knot_python_semantic/src/semantic_index/ast_ids.rs

```diff
@@ -34,20 +30,9 @@ pub(crate) struct AstIds {
 
     /// Maps expressions to their expression id. Uses `NodeKey` because it avoids cloning [`Parsed`].
     expressions_map: FxHashMap<ExpressionNodeKey, ScopedExpressionId>,
-
-    statements: IndexVec<ScopedStatementId, AstNodeRef<ast::Stmt>>,
-
-    statements_map: FxHashMap<NodeKey, ScopedStatementId>,
```

Comment:
We don't really need this anymore. Just storing them on the `Definition` and `NodeWithScope` is easier. It has the downside that we now create two `AstNodeRef` nodes for `Function`s and `Class`es. I think that's fine but we'll see. We can always go back to something more complicated if needed.